### PR TITLE
Devmode

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by lebauce** [comment]
run skydive-selenium-tests

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

File: contrib/dev/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  "analyzer.listen" => "0.0.0.0:8082",
+  "analyzer.storage.backend" => "elasticsearch",
+  "graph.backend" => "elasticsearch",
+  "storage.elasticsearch.host" => "192.168.10.10:9200"
+}
+
+$guest_additions_script = <<SCRIPT
+sudo dnf -y install https://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm
+sudo dnf -y install kernel kernel-devel VirtualBox-guest-additions
+latest=`rpm -q kernel-devel --last | head -n 1 | awk -F'kernel-devel-' '{print $2}' | cut -d " " -f 1`
```
Comment (nplanel) [inline contrib/dev/Vagrantfile]: rpm  -q kernel-devel --qf '%{BUILDTIME} %{version}-%{release}.%{arch}\n'  | tail -n 1 | awk '{print $2}'

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by hunchback** [comment]
@lebauce - rather than sync'ing `../..` (host) with `/home/vagrant/gopath/src/github.com/skydive-project/skydive` you may rather want to sync ~/gopath/src/github.com/skydive-project/skydive` (host), thus disconnecting the position from which we run `vagrant up` and the location of the source files on the host.

File: contrib/ansible/roles/elasticsearch/tasks/main0.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (safchain) [inline contrib/ansible/roles/elasticsearch/tasks/main0.yml]: I dont't understand the purpose of this main0 file, something commited by error ?

File: contrib/ansible/roles/skydive_dev/tasks/lxd.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+- selinux:
+    policy: targeted
+    state: permissive
+
+- shell: "dnf copr enable -y ganto/lxd"
```
Comment (safchain) [inline contrib/ansible/roles/skydive_dev/tasks/lxd.yml]: this is not going to work on centos

File: contrib/ansible/roles/skydive_dev/tasks/static.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+     - libxml2-static
+- shell: |
+    cd /tmp
+    git clone https://github.com/the-tcpdump-group/libpcap.git
+    cd libpcap
+    git checkout libpcap-1.5.3
+    ./configure --prefix=/usr/local --disable-shared --disable-dbus --disable-bluetooth --disable-canusb
+    make
+    sudo make install
+    cd ..
```
Comment (safchain) [inline contrib/ansible/roles/skydive_dev/tasks/static.yml]: nit: useless

File: contrib/ansible/roles/elasticsearch/tasks/main0.yml (hunk 2 above)
Comment (lebauce) [reply contrib/ansible/roles/elasticsearch/tasks/main0.yml]: Committed by error :-)

File: contrib/ansible/roles/skydive_dev/tasks/lxd.yml (hunk 3 above)
Comment (lebauce) [reply contrib/ansible/roles/skydive_dev/tasks/lxd.yml]: The COPR repository only supports Fedora. I'm disabling LXD for CentOS

File: contrib/ansible/roles/skydive_dev/tasks/static.yml (hunk 4 above)
Comment (lebauce) [reply contrib/ansible/roles/skydive_dev/tasks/static.yml]: Removed

**Comment by lebauce** [comment]
@hunchback I would prefer not to have to set GOPATH to be able to use this Vagrantfile

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by lebauce** [comment]
run skydive-unit-tests

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

File: scripts/ci/ansible/deploy.yml [hunk 5]
```diff
@@ -145,32 +84,12 @@
...
-      cd /tmp
-      git clone https://github.com/the-tcpdump-group/libpcap.git
-      cd libpcap
-      git checkout libpcap-1.5.3
-      ./configure --prefix=/usr/local --disable-shared --disable-dbus --disable-bluetooth --disable-canusb
-      make
-      sudo make install
-      cd ..
+  roles:
+    - role: skydive_dev
```
Comment (nplanel) [inline scripts/ci/ansible/deploy.yml]: role skydive_dev is in contrib/ansible/roles
how it would be included ?

Comment (lebauce) [reply scripts/ci/ansible/deploy.yml]: We can use ANSIBLE_ROLES_PATH variable. Or we can simply create a symlink

Comment (nplanel) [reply scripts/ci/ansible/deploy.yml]: could you add/set it in  scripts/ci/ansible/run-playbook.sh

**Comment by nplanel** [comment]
run skydive-selenium-tests

**Comment by nplanel** [comment]
run skydive-unit-tests

**Comment by nplanel** [comment]
run skydive-selenium-tests

**Pull request merged**